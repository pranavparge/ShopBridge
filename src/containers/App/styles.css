main {
  margin: 2em 1em 0em 1em;
}

.adminLink {
  text-decoration: none;
  color: unset;
  font-size: large;
  font-weight: 500;
  padding: 1em;
}

.adminactive {
  text-decoration: underline;
  color: #0d6efd;
  font-size: large;
  font-weight: 500;
  padding: 1em;
}

.adminLink:hover {
  text-decoration: underline;
}
