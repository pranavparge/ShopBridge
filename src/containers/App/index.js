import React from "react";
import { Switch, Route, Link, useLocation } from "react-router-dom";

import "./styles.css";

import Dashboard from "../Dashboard";
import Brand from "../../components/Brand";
import { Container, Nav, Navbar } from "react-bootstrap";

function App() {
  const location = useLocation();

  return (
    <div className="App">
      <header>
        <Navbar bg="light" variant="light">
          <Container>
            <Link
              to="/"
              className={
                (location.pathname === "/" && "adminactive") || "adminLink"
              }
            >
              ShopBridge
            </Link>

            <Nav className="me-auto">
              <Link
                to="/admin"
                className={
                  (location.pathname === "/admin" && "adminactive") ||
                  "adminLink"
                }
              >
                Admin
              </Link>
            </Nav>
          </Container>
        </Navbar>
      </header>

      <main>
        <Switch>
          <Route exact path="/">
            <Brand />
          </Route>
          <Route exact path="/admin">
            <Dashboard />
          </Route>
        </Switch>
      </main>
    </div>
  );
}

export default App;
