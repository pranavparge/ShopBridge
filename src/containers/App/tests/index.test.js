import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../index";

const component = (
  <BrowserRouter>
    <App />
  </BrowserRouter>
);

test("to check if it matches snapshot", () => {
  const { asFragment } = render(component);
  expect(asFragment(component)).toMatchSnapshot();
});

test("to check if admin console renders", () => {
  render(component);
  fireEvent.click(screen.getByTestId("adminLink"));
  expect(screen.getByText("Inventory")).toBeInTheDocument();
});

test("to check if home screen renders", () => {
  render(component);
  fireEvent.click(screen.getByTestId("adminLink"));
  fireEvent.click(screen.getByTestId("homeLink"));

  expect(screen.getByText("Welcome to")).toBeInTheDocument();
  expect(screen.getAllByText("ShopBridge")[0]).toBeInTheDocument();
  expect(screen.getAllByText("ShopBridge")[1]).toBeInTheDocument();
});
