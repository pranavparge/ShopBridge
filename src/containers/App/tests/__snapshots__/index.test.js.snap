// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to check if it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header>
      <nav
        class="navbar navbar-expand navbar-light bg-light"
      >
        <div
          class="container"
        >
          <a
            class="adminactive"
            data-testid="homeLink"
            href="/"
          >
            ShopBridge
          </a>
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="adminLink"
              data-testid="adminLink"
              href="/admin"
            >
              Admin
            </a>
          </div>
        </div>
      </nav>
    </header>
    <main>
      <div
        class="banner"
      >
        <h1>
          Welcome to 
          <span
            class="brand"
          >
            ShopBridge
          </span>
        </h1>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
